bin/src/MixinGenerator.php

```diff
@@ -0,0 +1,436 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Webmozart\Assert\Bin;
+
+use InvalidArgumentException;
+use ReflectionClass;
+use ReflectionException;
+use ReflectionMethod;
+use RuntimeException;
+use Webmozart\Assert\Assert;
+
+final class MixinGenerator
+{
+    /**
+     * @psalm-var list<string>
+     *
+     * @var string[]
+     */
+    private $unsupportedMethods = [
+        'nullOrNotInstanceOf',  // not supported by psalm (https://github.com/vimeo/psalm/issues/3443)
+        'allNotInstanceOf',     // not supported by psalm (https://github.com/vimeo/psalm/issues/3443)
+        'nullOrNotEmpty',       // not supported by psalm (https://github.com/vimeo/psalm/issues/3443)
+        'allNotEmpty',          // not supported by psalm (https://github.com/vimeo/psalm/issues/3443)
+        'allNotNull',           // not supported by psalm (https://github.com/vimeo/psalm/issues/3443)
+        'nullOrIsNotA',         // not supported by psalm (https://github.com/vimeo/psalm/issues/3444)
+        'allIsNotA',            // not supported by psalm (https://github.com/vimeo/psalm/issues/3444)
+        'nullOrNotFalse',       // not supported by psalm (https://github.com/vimeo/psalm/issues/3443)
+        'allNotFalse',          // not supported by psalm (https://github.com/vimeo/psalm/issues/3443)
+        'nullOrUpper',          // not supported by psalm (https://github.com/vimeo/psalm/issues/3443)
+        'allUpper',             // not supported by psalm (https://github.com/vimeo/psalm/issues/3443)
+        'nullOrIsNonEmptyList', // not supported by psalm (https://github.com/vimeo/psalm/issues/3444)
```

Comment:
https://github.com/vimeo/psalm/issues/3444 was declined, but I think the link as-is is a good reference (to be kept)